### PR TITLE
Add SignerWithAddress to hardhat-ethers

### PR DESCRIPTION
**Pull request by fvictorio**

### CODE REVIEW

File: packages/hardhat-ethers/src/helpers.ts [hunk 1]
```diff
@@ -6,6 +6,8 @@ import {
   NetworkConfig,
 } from "hardhat/types";
 
+import { SignerWithAddress } from "./signer-with-address";
```
Comment (alcuadrado) [inline packages/hardhat-ethers/src/helpers.ts:9]: Can we import this dynamically?

File: packages/hardhat-ethers/src/helpers.ts [hunk 2]
```diff
@@ -23,11 +25,23 @@ export interface FactoryOptions {
 
 const pluginName = "hardhat-ethers";
 
-export async function getSigners(hre: HardhatRuntimeEnvironment) {
+export async function getSigners(
+  hre: HardhatRuntimeEnvironment
+): Promise<SignerWithAddress[]> {
+  const { SignerWithAddress: SignerWithAddressImpl } = await import(
```
Comment (fvictorio) [inline packages/hardhat-ethers/src/helpers.ts:31]: I hate the `no-shadowed-variable` rule.

Let me know if you'd prefer to import this without aliasing and to import the type as `SignerWithAddressT` or something like that.

Comment (alcuadrado) [reply packages/hardhat-ethers/src/helpers.ts:31]: lgtm 🤷‍♂️

Comment (alcuadrado) [reply packages/hardhat-ethers/src/helpers.ts:31]: the rule is pretty annoying, i know

**Pull request merged**